src/Be.Vlaanderen.Basisregisters.MessageHandling.Kafka.Simple/KafkaConsumer.cs

```diff
@@ -26,37 +28,47 @@ public static class KafkaConsumer
 
             var serializer = JsonSerializer.CreateDefault(options.JsonSerializerSettings);
 
-            using var consumer = new ConsumerBuilder<Ignore, string>(config)
-                .SetValueDeserializer(Deserializers.Utf8)
-                .Build();
+            var consumerBuilder = new ConsumerBuilder<Ignore, string>(config)
+                .SetValueDeserializer(Deserializers.Utf8);
+            if (offset.HasValue)
+            {
+                consumerBuilder.SetPartitionsAssignedHandler((cons, topicPartitions) =>
+                {
+                    var partitionOffset = topicPartitions.Select(x => new TopicPartitionOffset(x.Topic, x.Partition, offset.Value));
+                    return partitionOffset;
+                });
+            }
+
+            using var consumer = consumerBuilder.Build();
             try
             {
                 consumer.Subscribe(topic);
 
+                var kafkaJsonMessage = new KafkaJsonMessage("", "");
                 while (!cancellationToken.IsCancellationRequested)
                 {
                     var consumeResult = consumer.Consume(TimeSpan.FromSeconds(3));
                     if (consumeResult == null) //if no message is found, returns null
                     {
-                        continue;
+                        break;
```

Comment:
What's the reason for this change?
With `continue` it kept retrying, because with first time connection it could timeout.

Reply:
There was also another reason: when the consumer reached offset 0 and polls for next message it will return `null`.
So with this change, once the consumer is up to date we stop listening.